deploy/charts/checkmk/templates/node-collector-container-metrics-ds.yaml

```diff
@@ -62,6 +62,9 @@ spec:
           args:
           {{- with .Values.nodeCollector.cadvisor.additionalArgs }}
             {{ toYaml . | nindent 12 }}
+          {{- if $.Values.k3s.enabled }}
```

Comment:
Why do you use {{- if $.Values.k3s.enabled }} instead of {{- if .Values.k3s.enabled }} ?
I am no Helm template expert, but it loosk rather unusual.

Reply:
You have to use `$` if you are inside a loop or iteration (as here introduced with `with` statement), in order to access variables on global level; because inside a loop using `.` you would access variables inside the local context.

But nonetheless, why are we adding it inside the context in the first place? For me this reads like, for each additionalArg we add, if k3s is enabled, also add `--containerd` ... but we would like to only add it once, no?

Reply:
Yes, it would be better to do this outside the context. 
We only need to add --containerd once, if the user is using k3s or rke2

Reply:
So it does just add it once and there was a weird thing I had happen to me where it actually added the other additionalArgs 2 times instead of once. The version you see here doesn't do that (at least for me). It just adds 1 containerd arg line after the original additionalArgs.